src/gateway/app.py

```
@@ -0,0 +1,117 @@
+    try:
+        size = int(length) if length else 0
+    except (TypeError, ValueError):
+        size = 0
+    body = environ.get("wsgi.input")
+    if not isinstance(body, io.BytesIO):
+        body_stream = io.BytesIO(body.read(size) if body else b"")  # type: ignore[attr-defined]
+    else:
+        body_stream = body
+    raw = body_stream.read(size or -1)
+    if not raw:
+        raise ValueError("empty body")
```

Ruff (0.14.2)

Avoid specifying long messages outside the exception class

(TRY003)